auth/oidc.go

```diff
@@ -439,3 +446,22 @@ func SetURLQueryParam(strURL, name, value string) (string, error) {
 	uri.RawQuery = rawQuery.Encode()
 	return uri.String(), nil
 }
+
+// GetHttpClient returns a new HTTP client with TLS certificate verification
+// disabled when insecureSkipVerify is true and enabled otherwise.
+func GetHttpClient(insecureSkipVerify bool) *http.Client {
+	if insecureSkipVerify {
+		transport := base.DefaultHTTPTransport()
+		transport.TLSClientConfig = &tls.Config{InsecureSkipVerify: true}
```

Comment:
This is probably technically ok since DefaultHTTPTransport doesn't have a TLSClientCOnfig, but it would be safer to check that transport.TLSClientConfig is nil, and update the existing if non-nil.  This would protect us from future changes to our DefaultHTTPTransport(), or http.DefaultTransport.  The standard library seems to follow the pattern:
```
	if transport.TLSClientConfig == nil {
		transport.TLSClientConfig = new(tls.Config)
	}
        transport.TLSClientConfig.InsecureSkipVerify = true

```

Reply:
Make sense, added changes to check whether the transport.TLSClientConfig is nil, and update the existing if non-nil.